### ISSUE TITLE
[Vite plugin] Port advanced SSR features and style handing from the current bundler

### ISSUE THREAD

**Issue by haikyuu** [imba/imba#718]
@somebee Can you list the SSR features needed to make the vite plugin on par with the current bundler?

**Comment by haikyuu** [comment]
The main thing is actually related to CSS, the goal is to not include duplicate CSS when we have components that live in both the server and the client.

**Comment by haikyuu** [comment]
This means that this is actually related to #716 and #715

**Comment by haikyuu** [comment]
There are some tests for this, which will make things easier to implement: 

https://github.com/imba/imba/blob/e9771ad3416c12e1f95081ab0a4e50c8d88e3a08/packages/imba/test/cli/examples/app/about.ssr.imba#L10

https://github.com/imba/imba/blob/e9771ad3416c12e1f95081ab0a4e50c8d88e3a08/packages/imba/test/cli/examples/app/css.imba#L2

These are css variables that are used to test the value they hold depending on their rendering
https://github.com/imba/imba/blob/e9771ad3416c12e1f95081ab0a4e50c8d88e3a08/packages/imba/test/cli/index.imba#L35-L40

**Comment by haikyuu** [comment]
The only feature that was left off is server side styles:
```ruby
import express from 'express'

global css html bg:amber7

#...
app.get "*", do(req, res)
  res.send String <App>
```
This would actually include the styles in the css rendered in the client.

Closing as I'm thinking of leaving this off. If you need this feature, please open a new issue
Closed with #727
